src/main/java/uk/gov/companieshouse/company/links/service/CompanyInsolvencyService.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.companieshouse.company.links.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.InternalApiClient;
+import uk.gov.companieshouse.api.http.ApiKeyHttpClient;
+import uk.gov.companieshouse.api.http.HttpClient;
+import uk.gov.companieshouse.api.insolvency.CompanyInsolvency;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.company.links.exception.RetryableErrorException;
+import uk.gov.companieshouse.logging.Logger;
+
+@Service
+public class CompanyInsolvencyService extends BaseApiClientService {
+
+    @Value("${api.company-insolvency-api-key}")
+    private String companyInsolvencyApiKey;
+
+    @Value("${api.endpoint}")
```

Comment:
property key should be the new one you introduced instead of "api.endpoint"?

Reply:
Relevant code change done